lib/ansible/modules/cloud/opennebula/one_vm.py

```diff
@@ -1300,15 +1300,14 @@ def get_connection_info(module):
     if not username:
         if not password:
             authfile = os.environ.get('ONE_AUTH')
-            if authfile is not None:
-                try:
-                    authstring = open(authfile, "r").read().rstrip()
-                    username = authstring.split(":")[0]
-                    password = authstring.split(":")[1]
-                except BaseException:
-                    module.fail_json(msg="Could not read ONE_AUTH file")
-            else:
-                module.fail_json(msg="No Credentials are set")
+            if authfile is None:
+                authfile = os.environ.get('HOME') + '/.one/one_auth'
+            try:
+                authstring = open(authfile, "r").read().rstrip()
+                username = authstring.split(":")[0]
+                password = authstring.split(":")[1]
+            except BaseException:
```

Comment:
```suggestion
            except FileNotFoundError:
            ...
            except Exception as e:
            ...
```
Don't use BaseException

Reply:
FileNotFoundError does not exist in python 2 , so I use OSError, and IOError exception. now the error output is like:

`localhost | FAILED! => {
    "changed": false,
    "msg": "Could not find or read ONE_AUTH file at '/home/ubuntu/.one/one_auth'"
}`

Reply:
I updated it to be `            except (OSError, IOError) as e:
                module.fail_json(msg=("Could not find or read ONE_AUTH file at '%s'" % authfile))` for compatibility with python 2